File: solutions/typescript/perfect-numbers/1/perfect-numbers.ts
export function classify(number: number): string {
  // Check if the number is positive
  if (number <= 0) {
    throw new Error('Classification is only possible for natural numbers.');
  }
  
  // Find the aliquot sum (sum of all proper divisors)
  let aliquotSum = 0;
  for (let i = 1; i <= number / 2; i++) {
    if (number % i === 0) {
      aliquotSum += i;
    }
  }
  
  // Classify the number based on its aliquot sum
  if (aliquotSum === number) {
    return 'perfect';
  } else if (aliquotSum > number) {
    return 'abundant';
  } else {
    return 'deficient';
  }
}

File: solutions/ruby/perfect-numbers/1/perfect_numbers.rb
class PerfectNumber
  def self.classify(number)
    # El mensaje debe ser EXACTAMENTE el que espera el test
    raise ArgumentError, 'Classification is only possible for positive integers.' if number < 1
    
    return 'deficient' if number == 1

    sum = 1 
    sqrt_n = Math.sqrt(number).to_i

    (2..sqrt_n).each do |i|
      if number % i == 0
        sum += i
        complement = number / i
        sum += complement if complement != i
      end
    end

    if sum == number
      'perfect'
    elsif sum > number
      'abundant'
    else
      'deficient'
    end
  end
end